### DISCUSSION TITLE
Qualcuno è riuscito ad integrare la PDC Templari Kita con Home Assistant via MODBUS RTU 485 ?

### DISCUSSION THREAD

**Discussion by xdiego76** [indomus/forum#1799] [HUB: Home Assistant]
Buongiorno,
mi chiedevo se nella community ci fosse qualche possessore di pompa di calore Templari che abbia integrato il loro sistema HCC (House Climate Control) ad Home Assistant attraverso il protocollo di comunicazione seriale MODBUS RTU 485.

Credo che la casa madre fornisca le tabelle descrittive dei registri dove è possibile comprendere cosa rappresentino, la lunghezza in termini di byte e quale trasformazione/scaling richiedano. Se così fosse credo dovrebbe esser possibile gestire completamente all'interno di HA non solo la PDC ma anche tutti gli altri componenti ad essa interfacciata.

Grazie, un saluto a tutti!
Diego

**Comment by bizzottogary** [comment]
Si, la mia l'ho collegata via RS485

**Reply by MicheleG84** [reply]
Ciao Gary. Integrare il sistema HCC di templari su Home Assistant attraverso Modbus interesserebbe pure a me.
Nel dettaglio mi chiedevo se era possibile integrare su Home Assistant i sensori room di Templari tramite l'entità "climates" in modo da leggere  temperatura - umidita delle stanze e se possibile, controllare i vari setpoint (ECO - Giorno - Notte // Riscaldamento & Raffrescamento) mentendo pero inalterato il controllo da K-Touch e dai vari sensori Room qualora fossi a casa...

**Reply by Atlantis1971** [reply]
Ciao,

anche io ho una pdc Templari. Via rete (considerato che il flusso dati già passa di lì) non è possibile fare niente? A me basterebbe il monitoraggio. Certo, se via RS485 si può anche agire sull'HCC è meglio!

Rimango in attesa.

Grazie molte!

Ale

**Comment by matteo710** [comment]
Interessante! Puoi dirmi qualcosa in più su come hai operato?

**Comment by melaqu** [comment]
Ciao, interesserebbe molto anche a me capire come hai fatto, mi potresti dare qualche informazione? Grazie

**Comment by wergio** [comment]
Seguo. Interesserebbe anche a me trovare un modo! a quanto pare esiste da un anno circa un software 1.9 del pannello di controllo HCC che alcuni utenti hanno in beta, ma non rilasciato pubblicamente che permette modbus over TCP. Ma per qualche strano motivo Templari non vuole rilasciarlo pubblicamente... per ora l'unico modo per ottenere i dati in sola lettura sono soluzioni simili a quella proposta da k7d...

**Reply by danros3** [reply]
Ma tu hai trovato un post / guida su questa integrazione? Io non ho trovato nulla e quindi non riesco a usarla.
Grazie

**Comment by Atlantis1971** [comment]
Ciao a tutti,

condivido con voi questo pdf, spero sia utile a chi di dovere.

Chiedo: per interfacciarsi con la pdc e controllarla non basterebbe "fare finta" di essere un termostato che si connette alle loro uscite floor?

[manuale-HCC-1075-ITA-17_09_2021.pdf](https://github.com/user-attachments/files/19111300/manuale-HCC-1075-ITA-17_09_2021.pdf)

**Reply by wergio** [reply]
se interessa la sola lettura è quello che fa k7d come descrive nel suo post, ma non è la soluzione ideale perchè è pur sempre un accrocchio, e richiede di collegare un nuovo elemento hardware all'interno della catena modbus senza fare danni, sicuramente fattibile ma non ideale. 

se può interessare allego un lavoro simile che era stato fatto da un'altra persona che l'aveva postato sul gruppo facebook templari, ma richiede anche l'uso di nodered per trasformare il modbus in mqtt
[Reverse Engineering MODBUS.docx](https://github.com/user-attachments/files/19112286/Reverse.Engineering.MODBUS.docx)

la cosa più semplice sarebbe che Templari esponesse semplicemente il modbus (o l'mqtt) sull'ip del pannello touch... da quello che ho capito la cosa è già supportata ma solo per loro utilizzo interno, e non viene permesso al cliente di prelevare questi dati (parlo del software disponibile al momento, si spera che futuro sia aperto a tutti... anche se è ormai un anno che se ne parla...)

**Reply by Atlantis1971** [reply]
Daniele hai gli estremi dell'autore del documento?

**Reply by wergio** [reply]
questo è il post originale su gruppo FB templari puoi provare a contattarlo con un messaggio privato
https://www.facebook.com/groups/possessori.templari/posts/1430901074410125/

PS: in vari abbiamo pressato in quel gruppo (dove ci sono anche tecnici Templari) da anni per avere un'apertura ufficiale su modbus tcp o mqtt, ma per ora nulla...

**Comment by matteo710** [comment]


**Reply by melaqu** [reply]
Non sapevo ci fosse su haks, io l’ho installato manualmente

**Comment by matteo710** [comment]
Come si fa ad installarlo manualmente?Il 8 Mar 2025 18:34, melaqu ***@***.***> ha scritto:
Non sapevo ci fosse su haks, io l’ho installato manualmente

**Reply by melaqu** [reply]
Vai su https://github.com/k7d/ha-templari-kita
Clic su “Code” e poi su “Download ZIP”
Salvi la cartella sul tuo computer, le decomprimi, apri la cartella /ha-templari-kita-main/custom_components
Copia la cartella “templari_kita” e la incolli nella directory del tuo HA /config/custom_components. Io per trasferire i file uso ftp come componente aggiuntivo su HA

**Reply by Atlantis1971** [reply]
Scusate ma su Hacs non ho trovato niente di Templari...

Potrei sapere come cercarlo per favore?

**Reply by melaqu** [reply]
Su haks non c’è, devi aggiungere manualmente la repository però, a quanto pare, a qualcuno non funziona. L’installazione manuale a me ha funzionato, le indicazioni su come fare le trovi nel mio post precedente

**Comment by matteo710** [comment]
Devi inserirlo come archivio digitale personalizzato, incollando il link githubIl 10 Mar 2025 11:17, Atlantis1971 ***@***.***> ha scritto:
Scusate ma su Hacs non ho trovato niente di Templari...
Potrei sapere come cercarlo per favore?

**Reply by Atlantis1971** [reply]
Perfetto, grazie. Ora mi devo prendere il prodottino su Amazon...

Certo se si potessero condividere gli sforzi sarebbe bellissimo! Tipo che ognuno fa un pezzo...

**Reply by danros3** [reply]
Ciao a tutti.
Io l'ho installato, ho l'apparecchio rs485->tcp con ip 192.168.x.x.
Con la sua interfaccia web l'ho configurato, mi pare a dovere.

Quando nell'integrazione templari provo mi chiede IP e porta dell'host ma, dopo aver inserito i dati mi risponde sempre con "host non raggiungibile"
Qualcuno ci è riuscito e magari spiega che dati inserire?
Io probabilmente canno qualcosa li.

Grazie

**Reply by melaqu** [reply]
![IMG_1654](https://github.com/user-attachments/assets/1c2dc548-438c-4a7a-bf17-2721d9fd76ee)

questa è la mia configurazione, a me funziona perfettamente

**Comment by Atlantis1971** [comment]
Ciao ragazzi,

ho parlato recentemente con Templari ed ho saputo che prossimamente unificheranno il software a bordo dell'hcc quindi tutti avremo lo stesso sw.
La novitrà vera è che sarà disponibile una modalità di esportazione dei dati di funzionamento della pdc e, mi è sembrato di aver capito, ci si potrà collegare ufficialmente tramite modbus.

Appena avrò news, vi informerò.

Ale

**Reply by wergio** [reply]
ottimo, si avevo visto anche io un messaggio sul gruppo facebook ufficiale, dove una persona dell'azienda aveva detto che entro un mese sarebbe stato abilitato il modbus over tcp in sola lettura, ma poi è stato cancellato mi sa, perchè non c'è più adesso, speriamo lo facciano per davvero! Già avere la lettura sarebbe un gran passo avanti

**Reply by Atlantis1971** [reply]
Sì, appena sarà possibile inizierò ad archiviare ed analizzare i dati di funzionamento per ottimizzare la configurazione e monitorare eventuali malfunzionamenti. Non vedo l'ora!

Prossimamente mi prenderò il baracchino modbus, per portarmi avanti.

**Reply by wergio** [reply]
per caso hai avuto poi notizie sul software hcc con supporto modbus?! ma temo che non abbiano poi rilasciato nulla... la cosa assurda è che i display HCC sono dei PLC della Weintek (il 10" che ho anche io è un Weintek MT8090XE) che supporterebbero il modbus tcp con un clic se lo volessero... veramente inspiegabile questa loro "chiusura" senza senso...

**Reply by Atlantis1971** [reply]
Ciao,
sfortunatamente non ho novità a riguardo ma appena passano le vacanze torno all'attacco! :-)

**Reply by Casamia2020** [reply]
> per caso hai avuto poi notizie sul software hcc con supporto modbus?! ma temo che non abbiano poi rilasciato nulla... la cosa assurda è che i display HCC sono dei PLC della Weintek (il 10" che ho anche io è un Weintek MT8090XE) che supporterebbero il modbus tcp con un clic se lo volessero... veramente inspiegabile questa loro "chiusura" senza senso...

Anch'io ho quel display. Sono 3 anni che chiedo questa opzione e mi hanno anche risposto male perche' insistevo. Comunque internamente l'opzione esiste , e' solo disattivata. Ho visto i grafici che loro hanno in azienda.

**Reply by wergio** [reply]
la questione si sta nuovamente scaldando sul gruppo facebook se volete dare un'occhiata https://www.facebook.com/groups/possessori.templari/posts/2020704325429794/ in sostanza hanno detto che non hanno tempo di farlo perchè "hanno tanto altro da fare"... no comment...

**Reply by Casamia2020** [reply]
Due anni fa per aver fatto una richiesta mandando 2 email, mi hanno dato dello stalker. È che non vogliono farlo. L hanno detto anche chiaramente una volta.... Altrimenti gli utenti si fanno delle paranoie sui dati. Il fatto che si sono montati la testa, all inizio erano disponibili.

**Reply by wergio** [reply]
ATTENZIONE notizia bomba: sembra che dopo gli ultimi post si siano decisi a dare la funzionalità! vedete https://www.facebook.com/groups/possessori.templari/posts/2020704325429794/?comment_id=2022532978580262  e provate a scrivere per avere l'aggiornamento

dico sembra perchè confermo che mi hanno aggiornato il pannello alla versione 1.8.49 ma non ho capito esattamente come si dovrebbe attivare il modbus tcp sto facendo alcuni test se scopro qualcosa vi dico

**Reply by gnoto** [reply]
Buonasera, come hai fatto a chiedere l'aggiornamento? A me interesserebbe semplificare un po' e usare direttamente il modus esposto dal pannello. Grazie!

**Reply by wergio** [reply]
> Buonasera, come hai fatto a chiedere l'aggiornamento? A me interesserebbe semplificare un po' e usare direttamente il modus esposto dal pannello. Grazie!

leggi nel link che ho riportato dicono di scrivere a updates@templari.com per ricevere l'aggiornamento del pannello. a me l'hanno già fatto perchè ero intervenuto nella discussione su facebook dicendo che mi serviva. però devo dire che pur avendo l'aggiornamento non vedo ancora modo per attivare il modbus... proverò lunedì a chiedergli come fare...

**Reply by gnoto** [reply]
Bene, grazie. Proverò!

**Reply by gnoto** [reply]
Se hanno attivato la funzione Modbus TCP Server sul pannello Weintek, basta interrogare direttamente l'HCC via rete lan.

**Reply by Atlantis1971** [reply]
Ciao, rieccomi.
Ho recentemente interagito sul gruppo di fb, come alcuni di voi avranno visto. Alle fine dei conti pare sia fattibile, almeno a detta di Massimo Masiero, avere almeno i dati in lettura tramite modbus. Nei prossimi giorni sentirò il supporto tecnico di Templari per capire, nel concreto, come fare.

Ci aggiorniamo.

Ale

**Reply by wergio** [reply]
ho fatto varie prove ma non riesco a rilevare nessun sensore modbus anche con la nuova versione 1.8.49 del pannello che mi hanno installato, da quello che so alle poche persone a cui era stato messo un software beta appariva un menu "supervisor" dove si potevano settare varie cose, cosa non presente in questa versione che mi hanno installato... fatemi sapere se riuscite a capirci qualcosa quando ve lo installano... NOTA: con l'aggiornamento vi cambieranno la pass del VNC a 111111 se era settata diversamente, senza dirvi nulla. purtroppo è difficilissimo farsi rispondere chiaramente da loro... altra cosa che servirebbe sapere con certezza è la porta dove ascoltare e l'elenco esatto dei registri visto che non esiste una "discovery" per rilevarli in automatico, ad esempio io sto provando la seguente conf che però non rileva nulla.
la porta dovrebbe essere la 8000 e il registro della temperatura acs il 1002, ma tutto questo ottenuto per vie traverse e non certo...

```
modbus:
  - name: pdc_templari_pt
    type: tcp
    host: 192.168.1.155
    port: 8000
    sensors:
      - name: pdc_templari_pt_t_acs
        unit_of_measurement: °C
        slave: 1
        address: 1002
        scale: 0.1
        offset: 0
        precision: 1
        input_type: input
        data_type: int16
```

**Reply by Atlantis1971** [reply]
cercherò di avere notizie...

**Reply by Casamia2020** [reply]
> ho fatto varie prove ma non riesco a rilevare nessun sensore modbus anche con la nuova versione 1.8.49 del pannello che mi hanno installato, da quello che so alle poche persone a cui era stato messo un software beta appariva un menu "supervisor" dove si potevano settare varie cose, cosa non presente in questa versione che mi hanno installato... fatemi sapere se riuscite a capirci qualcosa quando ve lo installano... NOTA: con l'aggiornamento vi cambieranno la pass del VNC a 111111 se era settata diversamente, senza dirvi nulla. purtroppo è difficilissimo farsi rispondere chiaramente da loro... altra cosa che servirebbe sapere con certezza è la porta dove ascoltare e l'elenco esatto dei registri visto che non esiste una "discovery" per rilevarli in automatico, ad esempio io sto provando la seguente conf che però non rileva nulla. la porta dovrebbe essere la 8000 e il registro della temperatura acs il 1002, ma tutto questo ottenuto per vie traverse e non certo...
> 
> ```
> modbus:
>   - name: pdc_templari_pt
>     type: tcp
>     host: 192.168.1.155
>     port: 8000
>     sensors:
>       - name: pdc_templari_pt_t_acs
>         unit_of_measurement: °C
>         slave: 1
>         address: 1002
>         scale: 0.1
>         offset: 0
>         precision: 1
>         input_type: input
>         data_type: int16
> ```

Qualche anno fa c'è chi era riuscito a sniffare i registri modbus. Aveva spiegato anche come fare. Ma è un lavoro immenso. Io ho provato poi mi sono arreso. E anche rotto le palle di questi atteggiamenti insensati.

**Reply by wergio** [reply]
capisco la frustrazione, ma ci siamo molto vicini penso... scrivete tutti per ricevere l'update poi un modo lo troviamo...

**Reply by Casamia2020** [reply]
A me 2 anni fa avevano aggiornato il pannello alla versione 1.9, che ha la possibilità tramite la porta 8000 di interfacciarsi al modbus. Ma non ci sono mai riuscito anche perché non mi hanno dato istruzioni di come si fa.

**Reply by wergio** [reply]
> A me 2 anni fa avevano aggiornato il pannello alla versione 1.9, che ha la possibilità tramite la porta 8000 di interfacciarsi al modbus. Ma non ci sono mai riuscito anche perché non mi hanno dato istruzioni di come si fa.

allora ti manca veramente molto poco: devi cercare il menu "supervisor" che dovrebbe essere nel menu avanzate X, e da li abilitare il modbus. poi con la configurazione che ho postato sopra dovresti poter leggere il registro della temperatura ACS in home assistant, se ti funziona poi ti mando un elenco di altri registri che ho trovato, ma sono da provare

**Reply by Casamia2020** [reply]
![Screenshot_2025-09-25-15-33-11-693-edit_com weintek easyaccess](https://github.com/user-attachments/assets/f877d349-275b-483c-9048-64495af91c2b)
![Screenshot_2025-09-25-15-32-51-345-edit_com weintek easyaccess](https://github.com/user-attachments/assets/bfe296fc-4eab-47d2-aee3-bf3b3dbdaec1)

**Reply by Casamia2020** [reply]
Però dove c'è scritto abilita, non me lo lascia fare

**Reply by wergio** [reply]
quindi rimane sempre off anche se tenti di attivarlo? cmq mi viene il dubbio che quei parametri in quella schermata siano per la scrittura... se hai modo di provare ti suggerirei di provare comunque la configurazione che ho postato sopra con le dovute modifiche per il tuo ip, giusto per vedere se ti rileva qualcosa, in caso positivo significa che la lettura è abilitata

**Reply by Casamia2020** [reply]
Rimane on per un secondo poi torna off. Ho provato da cell la configurazione. Mi fa errore su modbus

**Reply by Casamia2020** [reply]
> quindi rimane sempre off anche se tenti di attivarlo? cmq mi viene il dubbio che quei parametri in quella schermata siano per la scrittura... se hai modo di provare ti suggerirei di provare comunque la configurazione che ho postato sopra con le dovute modifiche per il tuo ip, giusto per vedere se ti rileva qualcosa, in caso positivo significa che la lettura è abilitata

Grande. Ci sono riuscito dopo 2 riavvii. 
![Screenshot_2025-09-25-17-58-45-293-edit_io homeassistant companion android](https://github.com/user-attachments/assets/63fb5792-72c7-46a6-a30d-262d18b92002)
Se templari pubblicasse tutti gli indirizzi modbus

**Reply by wergio** [reply]
fantastico! quindi abbiamo la conferma che con la 1.9.0 funziona, io ho alcuni file con vari registri da provare, come te li posso mandare? 

non so perchè a me abbiano installato la 1.8.49 forse si sono sbagliati... mah...

**Reply by Casamia2020** [reply]
Ti ho inviato un email.
Quello che non capisco che ho sempre tentato di collegarmi con un modulo della Waveshare come templari indicava nelle misere istruzioni. E infatti puntavo all ip del modulo in questione. Invece non serve a niente come era prevedibile. E non serve neppure la pagina supervisire introdotta nella 1.9.

**Reply by wergio** [reply]
ok ti ho mandato tutto quello che ho, magari poi facci sapere che configurazione finale funziona a te. a quanto pare solo la 1.9 ha i registri esposti, nelle precedenti non è possibile in nessun modo tirare fuori dati, la porta 8000 è aperta ma non fornisce dati

**Reply by Casamia2020** [reply]
Grazie. Peccato che non posso passarvi la 1.9. Sono quasi 2 anni che me l hanno aggiornata. E all inizio non volevano. Ma non mi sarei mai aspettato che funzionasse. Prova a chiedere se è possibile avere quella versione.

**Reply by Atlantis1971** [reply]
Ecco, scusate, ho trovato anche io il menu supervisore con le impostazioni relative. Su Home Assistant ora cosa devo fare?

**Reply by wergio** [reply]
> Ecco, scusate, ho trovato anche io il menu supervisore con le impostazioni relative. Su Home Assistant ora cosa devo fare?

che versione del pannello hai 1.9? te l'hanno installata questi giorni o precedentemente? scrivimi a d.vergini@gameprog.it e ti mando quello che ho

**Reply by Casamia2020** [reply]
> Ecco, scusate, ho trovato anche io il menu supervisore con le impostazioni relative. Su Home Assistant ora cosa devo fare?

Io ci sono riuscito grazie a Daniele Vergini. Però  il supervisore non serve nel mio caso perché nemmeno riesco ad attivarlo. Mettendo i dati modbus come indicato sopra, home assistant trova i sensori. Ne ho messi 20. Allora molti non corrispondono, cioè bisogna guardare lo schermo ed abbinare gli indirizzi in base al dato. Qualcuno è a zero perché il riscaldamento è spento . Ma sono precisi rispetto all integrazione che era stata messa all inizio del post. Ci vuole pazienza e mettere tutto in ordine.

**Reply by gnoto** [reply]
Grazie per i contributi.
Ho capito che i dati sono esposti solo dalla versione 1.9.0 che io non ho. Pur avendo più volte chiesto un aggiornamento, non ho ricevuto risposta... no comment!
Chi ha la versione 1.9.0 può condividerla? Grazie!

**Reply by wergio** [reply]
purtroppo non è possibile condividere i software perchè non è possibile nemmeno prelevarli, teoricamente sarebbe fattibile con il software weintek easy builder, avevo fatto qualche esperimento tempo fa ma l'operazione è protetta da password che solo templari sa... quindi solo loro possono installare un nuovo software sul pannello, bisogna insistere ed essere molto pazienti. se vuoi ho creato un canale telegram dove chiaccherare delle varie soluzioni e novità mandami una mail se vuoi entrare

**Reply by gnoto** [reply]
@wergio Grazie, ti ho mandato l'indirizzo in privato

Riguardo il software del pannello HCC: io dal mio sono riuscito a prelevarlo con il software Weintek. Naturalmente, il codice che prelevi è protetto  e non editabile (altrimenti mi sarei abilitato il Modbus server da solo...), mo è instancabile su un altro pannello. Non so se questo dipende dalla mia versione, effettivamente vecchiotta.

**Comment by Casamia2020** [comment]
Ciao a tutti. Con la versione di Home Assistant 2025.9 l integrazione mi da errore pymodbus e non funziona più. Come avete risolto? Grazie

**Reply by wergio** [reply]
non sto usando l'integrazione ma purtroppo molte integrazioni di tutti i tipi stanno avendo problemi con pymodbus vedi ad esempio questa che spiega come sia necessario seguire le modifiche con release continue di aggiornamento per rimanere compatibili https://github.com/wlcrs/huawei_solar/releases/tag/1.5.7

**Reply by Casamia2020** [reply]
Adesso che avevo un minimo di controllo dei parametri. Templari non si muove in questo senso.

**Comment by wergio** [comment]
finalmente da qualche giorno è possibile chiedere l'aggiornamento del pannello HCC per avere il modbus/tcp! lo posso confermare perchè l'anno attivato a me e ad altre persone!
Chi è interessato può scrivere a updates@templari.com chiedete di avere l'aggiornamento "supervisore SVK". vi aggiorneranno il software del pannello, a me hanno messo la 1.8.55. ma non basta avere solo la versione aggiornata, devono proprio attivarvelo esplicitamente. noterete che sotto il menù avanzate X di fianco ad "altro" ci sarà il menù "supervisore" questo vi conferma che è attivo, non settare nulla nel menù ma semplicemente inserire la configurazione dei sensori nel configuration.yaml di home assistant come ad es la seguente, vi metto solo un registro di esempio (ma loro vi danno anche un elenco dei registri)
```
  - name: pdc_templari
    type: tcp
    host: 192.168.1.154
    port: 8000
    sensors:
      - name: pdc_templari_t_acs
        device_class: temperature
        slave: 1
        address: 2
        scale: 0.1
        offset: 0
        precision: 1
        input_type: input
        data_type: int16
``` 
PS: ho creato un gruppo telegram per discutere delle varie configurazioni se volete entrare mandatemi una mail, la trovate nel mio profilo

AGGIORNAMENTO: ho messo online su github i miei file di configurazione per configurare velocemente i vari registri modbus e usarli nell'interfaccia potete trovarli qui: https://github.com/wergio/templari-modbus-home-assistant/

**Reply by Fabio8819** [reply]
Confermo, richiesto oggi e me lo hanno fatto subito!
